File: beakjoon/11724/11724.py
import sys

sys.setrecursionlimit(10**8)

N, M = map(int, input().split())
visited = [False for _ in range(N+1)]
rel = [[] for _ in range(N+1)]

count = 0

for _ in range(M):
    a, b = map(int, sys.stdin.readline().split())
    rel[a].append(b)
    rel[b].append(a)


def dfs(num):
    visited[num] = True

    for r in rel[num]:
        if not visited[r]:
            dfs(r)


for i in range(1, N+1):
    if not visited[i]:
        dfs(i)
        count += 1
print(count)

File: beakjoon/11724/11724.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>

using namespace std;

int N, M;
bool visited[1001] = {false};
vector<int> graph[1001];

void dfs(int num)
{
    visited[num] = true;

    for (int i = 0; i < graph[num].size(); i++)
    {
        int N = graph[num][i];
        if (!visited[N])
            dfs(N);
    }
}
int main()
{
    cin >> N >> M;
    int a, b, count = 0;

    for (int i = 0; i < M; i++)
    {
        cin >> a >> b;
        graph[a].push_back(b);
        graph[b].push_back(a);
    }

    for (int i = 1; i < N + 1; i++)
    {
        if (!visited[i])
        {
            dfs(i);
            count++;
        }
    }
    cout << count << "\n";

    return 0;
}